astropy/modeling/physical_models.py

```diff
@@ -0,0 +1,190 @@
+            Invalid temperature.
+
+        ZeroDivisionError
+            Wavelength is zero (when converting to frequency).
+        """
+        if not isinstance(temperature, u.Quantity):
+            in_temp = u.Quantity(temperature, u.K)
+        else:
+            in_temp = temperature
+
+        # Convert to units for calculations, also force double precision
+        with u.add_enabled_equivalencies(u.spectral() + u.temperature()):
+            freq = u.Quantity(x, u.Hz, dtype=np.float64)
+            temp = u.Quantity(in_temp, u.K, dtype=np.float64)
+
+        # Check if input values are physically possible
+        if np.any(temp < 0):
+            raise ValueError(f"Temperature should be positive: {temp}")
+        if not np.all(np.isfinite(freq)) or np.any(freq <= 0):
+            warnings.warn(
+                "Input contains invalid wavelength/frequency value(s)",
+                AstropyUserWarning,
+            )
+
+        log_boltz = const.h * freq / (const.k_B * temp)
+        boltzm1 = np.expm1(log_boltz)
+
+        # Calculate blackbody flux
+        bb_nu = 2.0 * const.h * freq ** 3 / (const.c ** 2 * boltzm1) / u.sr
+
+        # strip the unit from the scale if it has one
+        if hasattr(scale, "unit"):
+            mult_scale = scale.value
+            bb_unit = scale.unit
+        else:
+            mult_scale = scale
+            bb_unit = u.erg / (u.cm ** 2 * u.s * u.Hz * u.sr)
+
+        y = mult_scale * bb_nu.to(bb_unit, u.spectral_density(freq))
+
+        # If the temperature parameter has no unit, we should return a unitless
+        # value. This occurs for instance during fitting, since we drop the
+        # units temporarily.
+        if hasattr(temperature, "unit"):
+            return y
+        else:
+            return y.value
+
+    @property
+    def input_units(self):
+        # The input units are those of the 'x' value, which should always be
+        # Hz. Because we do this, and because input_units_allow_dimensionless
+        # is set to True, dimensionless values are assumed to be in Hz.
+        return {"x": u.Hz}
+
+    def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
+        return OrderedDict([("temperature", u.K)])
+
+    # does not work as need to pass freq/wave for unit equivalencies
+    #    @property
+    #    def return_units(self):
+    #        if self.scale.unit is None:
+    #            return None
+    #        else:
+    #            return {'y': self.scale.unit}
+
+    @property
+    def bolometric_flux(self):
+        """Bolometric flux."""
+        return const.sigma_sb * self.temperature ** 4 / np.pi
+
+    @property
+    def lambda_max(self):
+        """Peak wavelength when the curve is expressed as power density."""
+        return const.b_wien / self.temperature
+
+    @property
+    def nu_max(self):
+        """Peak frequency when the curve is expressed as power density."""
+        return 2.8214391 * const.k_B * self.temperature / const.h
```

Comment:
Interesting. I thought you said exact calculation of this was not possible. What is 2.8214391? Is there some literature reference that is tied to this formula?

Reply:
Yep.  I put it in the docs.  It is https://en.wikipedia.org/wiki/Wien%27s_displacement_law